packages/data-stores/src/reader/mutations/use-site-unsubscribe-mutation.ts

```diff
@@ -107,9 +132,17 @@ const useSiteUnsubscribeMutation = () => {
 					context.previousSubscriptionsCount
 				);
 			}
+			if ( context?.previousSiteSubscriptionDetails ) {
+				queryClient.setQueryData(
+					siteSubscriptionDetailsCacheKey,
+					context.previousSiteSubscriptionDetails
+				);
+			}
 		},
 		onSettled: () => {
-			// pass in more minimal keys, everything to the right will be invalidated
+			// We are not invalidating the `siteSubscriptionDetailsCacheKey` on purpose here,
```

Comment:
It sounds not right to me. The react-query is meant to resemble backend state. Using it for frontend state management will potentially make it more difficult to maintain in the future.

Reply:
Thanks for your feedback, Maciej!

I think I have a better solution now, using `refetchType: 'none'` in the `invalidateQueries` call:

```js
queryClient.invalidateQueries( siteSubscriptionDetailsCacheKey, { refetchType: 'none' } );
```

Will test it out and if it works as expected, commit the change.

Reply:
This is now addressed in https://github.com/Automattic/wp-calypso/pull/77264/commits/ede480f83dcfd6bfc141ac7741cff5a0edfc37d5.